http/encoding.go

```diff
@@ -236,3 +245,45 @@ func SetContentType(w http.ResponseWriter, ct string) {
 	}
 	w.Header().Set("Content-Type", h+suffix)
 }
+
+func newTextHtmlEncoder(w io.Writer) Encoder {
+	return &textHtmlEncoder{w}
+}
+
+type textHtmlEncoder struct {
+	w io.Writer
+}
+
+func (e *textHtmlEncoder) Encode(v interface{}) error {
+	var b []byte
+
+	strptr, isstrptr := v.(*string)
+	if isstrptr {
+		b = []byte(*strptr)
+	} else {
+		b = []byte(v.(string))
+	}
+
+	_, err := e.w.Write(b)
+	return err
+}
+
+func newTextHtmlDecoder(r io.Reader) Decoder {
+	return &textHtmlDecoder{r}
+}
+
+type textHtmlDecoder struct {
+	r io.Reader
+}
+
+func (e *textHtmlDecoder) Decode(v interface{}) error {
+	strptr := v.(*string)
```

Comment:
We should also support slices of bytes. The code shouldn't panic if `v` is not a string nor a slice of bytes.